src/main/java/org/jenkinsci/plugins/github/internal/GitHubLoginFunction.java

```diff
@@ -107,31 +106,15 @@ private Proxy getProxy(String apiUrl) {
      * @return connector to be used as backend for client
      */
     private OkHttpConnector connector(GitHubServerConfig config) {
-        OkHttpClient client = new OkHttpClient().setProxy(getProxy(defaultIfBlank(config.getApiUrl(), GITHUB_URL)));
+        OkHttpClient.Builder builder = BASECLIENT.newBuilder()
```

Comment:
you can skip instance creation with builder creation directly via new OkHttpClient.Builder()

Reply:
@lanwen 
The guidance is to have a single client and then create new clients from that client using `newBuilder()`.